frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -33,22 +33,22 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
 }) => (
   <Form className="co-deploy-image" onSubmit={handleSubmit}>
     <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection />}
-    {values.type === EventSources.KafkaSource && <KafkaSourceSection />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
-    {values.type === EventSources.KafkaSource && <AdvancedSection />}
+    {eventSourceSectionLoader(values.type, namespace)}
+    {isKnownEventSourceSelected(values.type) && (
+      <>
+        <SinkSection namespace={namespace} />
+        <AppSection
+          project={values.project}
+          noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
+        />
+      </>
+    )}
+    {values.type === KnownEventSources.KafkaSource && <AdvancedSection />}
     <FormFooter
       handleReset={handleReset}
       errorMessage={status && status.submitError}
       isSubmitting={isSubmitting}
       submitLabel="Create"
-      sticky
```

Comment:
Why have you removed `sticky` here? I thought all of our forms need to have sticky form footer,

Reply:
@nemesis09 yes we need that

Reply:
it slipped while resolving merge conflicts. added `sticky`